File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        last = head
        curr = head.next
        sum = 0
        while curr:
            if curr.val == 0:
                last.next = curr
                last = curr
                last.val = sum
                sum = 0
            else:
                sum += curr.val
            curr = curr.next
        
        return head.next

File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode* temp = head->next;
        ListNode* lz = head;
        int val = 0;
        while(temp != NULL) {
            if(temp->val == 0) {
                temp->val = val;
                val = 0;
                lz->next = temp;
                lz = temp;
            } else {
                val += temp->val;
            }
            temp = temp->next;
        }
        return head->next;
    }
};